feat: blog: ESLint Community org announcement (#402)
eslint/eslint.org: src/content/drafts/eslint-community-org.md

  We took this one over from the
  [@nodesecurity](https://github.com/nodesecurity) team, but
  [@nzakas](https://github.com/nzakas) is still maintaining it
* [`eslint-utils`](https://github.com/eslint-community/eslint-utils) and
  [`regexpp`](https://github.com/eslint-community/regexpp)  
  We created a fork from [@mysticatea](https://github.com/mysticatea)'s repos and
  published them under the
  [`@eslint-community` npm organization](https://npmjs.com/org/eslint-community).

We also intend to fork, bring up to date, and publish the following packages
originally authored by [@mysticatea](https://github.com/mysticatea) as well:

* [`eslint-plugin-es`](https://github.com/mysticatea/eslint-plugin-es)  
  We'll work on getting all
  [`eslint-plugin-es-x`](https://github.com/eslint-community/eslint-plugin-es-x)
  features merged as well
* [`eslint-plugin-eslint-comments`](https://github.com/mysticatea/eslint-plugin-eslint-comments)
* [`eslint-plugin-node`](https://github.com/mysticatea/eslint-plugin-node)  
  We'll work on getting all
  [`eslint-plugin-n`](https://github.com/eslint-community/eslint-plugin-n)
  features merged as well

Of course, we don't want to stop there.

> If you think your project would be a good fit for the new community org, *or*

Reviewer: We don't need the extra emphasis here
```suggestion
If you think your project would be a good fit for the new community org or
```

you're depending on a project that's currently unmaintained and want to have the

> community to take it over, you can contact the Community Core team on

Reviewer: Kinder words :)

```suggestion
community to adopt it, you can contact the Community Core team on
```

> community to take it over, you can contact the Community Core team on

Reviewer: Please double-check capitalization throughout. In some places we are using "Community Core team" and in others "community core team". I'm fine with either, let's just be consistent.

> [the official ESLint Discord](https://eslint.org/chat).

Author: This could be a good idea 👍

> Keep in mind that we'll only be able to take over projects that are adhering to
> the following acceptance criteria:

Reviewer: A bit of cleanup here.

```suggestion
Keep in mind that we'll only be able to accept projects that adhere to the following criteria:
```

* Is it a package that is ESLint-related?  
  Mostly this will be ESLint plugins, but (unmaintained) dependencies of such

>   packages, closely related packages or packages split from the main ESLint repo

Reviewer 2: ```suggestion
  packages, closely related packages, or packages split from the main ESLint repo
```

  (like
  [`eslint-formatter-codeframe`](https://github.com/eslint-community/eslint-formatter-codeframe)
  or
  [`eslint-formatter-table`](https://github.com/eslint-community/eslint-formatter-table))
  or used by the main repo (like
  [`eslint-utils`](https://github.com/mysticatea/eslint-utils) and
  [`regexpp`](https://github.com/mysticatea/regexpp)) could go in the
  `@eslint-community` GitHub organization as well.

>   Because plugins can expose their own (recommended) configs, we already accept
>   config packages in some way.  
>   However, the community core team won't be eager to accept shared ESLint config
>   packages, as they can be very opinionated.

Reviewer: I think it's fine to say we just won't accept shareable configs.
```suggestion
  We won't be accepting shareable configs as these are typically very opinionated and are easy to extend when necessary.
```

>   It also doesn't matter if the package is purely TypeScript-related or not, as

Reviewer: I'm not sure I understand this sentence. Will we accept purely TypeScript-related packages or not? What type of fragmentation are we worried about?

Author: We **do** accept purely TS-related packages

If we wouldn't, that could mean a `typescript-eslint-community` organization could be created by the community, which will only fragmentize the community & that's not something we want

Reviewer: Okay, then I think we can just remove this sentence. I think the natural inclination will be to believe that any ESLint-related package can be a candidate, so I don't think we need to get this specific.

Author: Should all be fine now 👍

  we don't want to fragment the ESLint community more without any real benefit.

* Is it widely depended upon throughout the ESLint community?  

>   We don't have a real number in mind here, but the packages we currently took
>   over (or that we're planning to take over) almost all have at least 3M
>   downloads/week.

Reviewer: ```suggestion
  We don't have a real number in mind here, but the packages we currently maintain
  (or that we're planning to maintain) almost all have at least 3M
  downloads/week.
```

> * Is it currently unmaintained?  

Author: I took the wording we used in the RFC here

> * Is it currently unmaintained?  

Reviewer: The RFC was designed to get a starting point for the project, but we definitely discussed that this should be an org for more than just orphaned projects. There was a lot of discussion about who would be handling what if maintainers were still in the picture. If that's not what you're intending now, then we need to have a larger discussion because it seems that our two ideas for this org have diverged significantly and we need to fix that.

> * Is it currently unmaintained?  

Author: I don't mind having non-orphaned projects into the org, but I think I missed the discussions you are talking about 🤔
Hence why I only included the text about orphaned projects.

Feel free to fill me in on the outcome of them or point me to them if they're available publicly

> * Is it currently unmaintained?  

Reviewer: I mentioned this from the start:
https://github.com/eslint/eslint/discussions/15929#discussioncomment-2850464

I think we both probably weren't terribly clear when communicating this, as it seems like there are several places where you mentioned "unmaintained" whereas I was mentioning "in case they become unmaintained." And it seems like we never fully clarified our positions there.

There was also discussion on the RFC about inviting maintainers to continue working on the project as part of the org:

https://github.com/eslint/rfcs/pull/91#issuecomment-1145333115 - here, I was assuming that maintainers would still be maintaining their packages.
https://github.com/eslint/rfcs/pull/91#discussion_r899571650 - it seems like you were assuming the same here.

The RFC itself mentions, in several places, that not being maintained isn't a hard-and-fast rule:

https://github.com/eslint/rfcs/pull/91/files#diff-fe2bbc1acd365b32adaeb59fea2c39b87fb9974c30902c0342ecddfa13355138R110
https://github.com/eslint/rfcs/pull/91/files#diff-fe2bbc1acd365b32adaeb59fea2c39b87fb9974c30902c0342ecddfa13355138R157

So, I think we were probably just talking past each other a bit throughout the whole process. It happens, not a big deal.

What I would like to say going forward is that projects that meet the criteria besides no updates will be accepted. As we've learned through mysticatea's plugins, trying to regain control of plugins after they have fallen out of maintenance is a lot of work. It would be a lot easier if those plugins were already in eslint-community, being maintained by their original maintainers, and then we could step in if something happened to them. That's exactly why I transferred eslint-plugin-security and why I would like other plugin authors to do the same.

How does that sound?

>   We don't want to take over packages that are still maintained by their
>   original author(s), as that would be a waste of time for everyone involved.

Reviewer: I think we can eliminate this now?

```suggestion
```

> These criteria aren't written in stone, but they're good guidelines to see if we
> should take over a package or not.

Reviewer: ```suggestion
These criteria aren't written in stone, but are the guidelines we're beginning with to determine whether or not to accept a project into the organization.
```

> If we decide to take over a package, but we can't get into contact with the
> owner of the repo to transfer ownership over to the new GitHub organization, we
> will (temporarily) fork the repo and publish it under the `@eslint-community` npm
> organization. However, the ultimate goal will always be to transfer the original
> repo.

Reviewer: This reads as very aggressive, and I think we should just eliminate it altogether.
```suggestion
```

If you have any further questions, feel free to ask them on

> [the official ESLint Discord](https://eslint.org/chat).

Reviewer: We still need a conclusion at the end of this post.

Author: How would you see something like that?

Reviewer: A paragraph that summarizes what has already been discussed in the post. There are a lot of posts on the current blog you can look at for reference under the "Announcements" and "Storytime" categories.

Reviewer: ```suggestion
[the official ESLint Discord](https://eslint.org/chat/eslint-community).
```

---
layout: post
title: Announcing the ESLint Community GitHub organization
teaser: >-
  The ESLint Community GitHub organization is a place for important ecosystem
  projects to get the attention and support they deserve.
tags:
  - eslint-community
  - Updates
authors:
- MichaelDeBoey
categories:
  - Announcements
---

In our
[2022 year in review](https://eslint.org/blog/2023/01/eslint-2022-year-review),

> we briefly mentioned a couple of project kickoffs, and I'm excited to now

Reviewer 3: We should consistently use "we" as the voice.

```suggestion
we briefly mentioned a couple of project kickoffs, and we're excited to now
```

[donating to community projects](https://eslint.org/blog/2022/02/paying-contributors-sponsoring-projects/#supporting-the-community),
to ensure that high-value projects were receiving the funds they needed to
continue with ongoing maintenance and development. Our next step is to create
the `@eslint-community` GitHub organization as a home for high-value projects in
the ESLint ecosystem.

As you can read in the
["`@eslint-community` GitHub organization" RFC](https://github.com/eslint/rfcs/tree/main/designs/2022-community-eslint-org),
the goal of this new organization is to have a place where community members can
help ensure widely depended upon ESLint-related packages live and never fall out
of maintenance.  
The Community Core team (which currently consists of
[`@aladdin-add`](https://github.com/aladdin-add),
[`@ota-meshi`](https://github.com/ota-meshi),
[`@voxpelli`](https://github.com/voxpelli) & myself
[`@MichaelDeBoey`](https://github.com/MichaelDeBoey)) & all package maintainers
will make sure these packages stay up-to-date with newer ESLint releases, and
they don't hold the wider community back by depending on a single person's

> GitHub or `npm` account.

Reviewer 4: ```suggestion
GitHub or npm Registry account.
```

You might have noticed that some of your most beloved ESLint-related packages

> have already been moved over to or been forked by to the `eslint-community`

Reviewer 3: ```suggestion
have already been moved over to or been forked to the `eslint-community`
```

organization. A full list of the current projects include:

* [`eslint-formatter-codeframe`](https://github.com/eslint-community/eslint-formatter-codeframe)
  &
  [`eslint-formatter-table`](https://github.com/eslint-community/eslint-formatter-table)  
  We took these over from [@fregante](https://github.com/fregante)
* [`eslint-plugin-eslint-plugin`](https://github.com/eslint-community/eslint-plugin-eslint-plugin)  
  [@not-an-aardvark](https://github.com/not-an-aardvark) transferred this repo

>   to the new org, but is still maintaining it

Reviewer 5: just want to point out: @bmish has been the active maintainer for a while. 😄

Reviewer 3: Thanks @aladdin-add! Appreciate your support with it. Yes, I've been the primary maintainer on eslint-plugin-eslint-plugin for 2 years.

* [`eslint-plugin-promise`](https://github.com/eslint-community/eslint-plugin-promise)  
  We took this one over from [@xjamundx](https://github.com/xjamundx)
* [`eslint-plugin-security`](https://github.com/eslint-community/eslint-plugin-security)  
  We took this one over from the
  [@nodesecurity](https://github.com/nodesecurity) team, but
  [@nzakas](https://github.com/nzakas) is still maintaining it
* [`eslint-utils`](https://github.com/eslint-community/eslint-utils) &
  [`regexpp`](https://github.com/eslint-community/regexpp)  
  We created a fork from [@mysticatea](https://github.com/mysticatea)'s repos &
  published them under the

>   [`@eslint-community` `npm` organization](https://npmjs.com/org/eslint-community).

Reviewer 4: ```suggestion
  [`@eslint-community` npm Registry organization](https://npmjs.com/org/eslint-community).
```

of maintenance.  
The Community Core team (which currently consists of
[`@aladdin-add`](https://github.com/aladdin-add),
[`@ota-meshi`](https://github.com/ota-meshi),
[`@voxpelli`](https://github.com/voxpelli) & myself
[`@MichaelDeBoey`](https://github.com/MichaelDeBoey)) & all package maintainers
will make sure these packages stay up-to-date with newer ESLint releases, and
they don't hold the wider community back by depending on a single person's
GitHub or npm Registry account.

You might have noticed that some of your most beloved ESLint-related packages
have already been moved over to or been forked by the `eslint-community`
organization. A full list of the current projects include:

* [`eslint-formatter-codeframe`](https://github.com/eslint-community/eslint-formatter-codeframe)
  &
  [`eslint-formatter-table`](https://github.com/eslint-community/eslint-formatter-table)  
  We took these over from [@fregante](https://github.com/fregante)
* [`eslint-plugin-eslint-plugin`](https://github.com/eslint-community/eslint-plugin-eslint-plugin)  
  [@bmish](https://github.com/bmish) &
  [@not-an-aardvark](https://github.com/not-an-aardvark) transferred this repo

>   to the new org, but they're still maintaining it

Reviewer 3: Can we mention @aladdin-add for eslint-plugin-eslint-plugin too?
